js/ui/components/clubhouse.js

```diff
@@ -716,21 +723,48 @@ var ClubhouseOpenButton = new Lang.Class({
         this._highlightSoundItem = new SoundServer.SoundItem('clubhouse/entry/pulse');
         this._highlightLoopingSoundItem =
             new SoundServer.SoundItem('clubhouse/entry/pulse-loop');
+
+        this.connect('notify::visible',
+                     this._play_stop_highlighted_sound.bind(this));
+        this.connect('notify::highlighted',
+                     this._play_stop_highlighted_sound.bind(this));
     },
 
-    setHighlighted: function(highlighted) {
+    set highlighted(highlighted) {
         if (highlighted) {
             this.child = this._pulseIcon;
             this._pulseAnimation.play();
-            this._highlightSoundItem.play();
-            this._highlightLoopingSoundItem.play();
         } else {
             this.child = this._normalIcon;
             this._pulseAnimation.stop();
+        }
+        this._highlighted = highlighted;
+        this.notify('highlighted');
+    },
+
+    get highlighted() {
+        return this._highlighted;
+    },
+
+    _play_stop_highlighted_sound() {
+        const states = [
+            SoundServer.SoundItemStatusEnum.NONE,
+            SoundServer.SoundItemStatusEnum.PENDING,
+            SoundServer.SoundItemStatusEnum.CANCELLING
+        ];
+
+        if (this.visible && this.highlighted) {
+            // If the sound item has already an UUID assigned, then do not
```

Comment:
"UUID assigned" is an implementation detail of the sound server. There are no UUIDs here. I would keep the comment with the method names used here, or remove it.